src/golang/cmd/server/middleware/usage/usage.go

```diff
@@ -0,0 +1,86 @@
+package usage
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/denisbrodbeck/machineid"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	obfuscated string = "***"
+	delimiter  string = "/"
+	hashKey    string = "aqueduct"
+	// This is the Grafana Loki server address
+	logURL string = "http://34.25.83.71:9090/loki/api/v1/push"
```

Comment:
I don't know how this works in other open-source projects, but is it dangerous to expose the address of our metrics server publicly? It doesn't seem like pushing to this needs any credentials, so would anyone be able to flood Loki with requests?

Reply:
Yeah this is a good question! On the Loki side we can implement policies to block random requests that don't fit our current schema, but currently there is no way to prevent an attacker from faking a bunch of valid log entries and sending them to our server. I'm looking into how other projects are addressing this.